Add old upgrade guide, update it and integrate with new chapters

pbauer opened the pull request.

pbauer committed:
Add old upgrade guide and integrate with new chapters

pbauer committed:
Add old upgrade guide and integrate with new chapters

pbauer committed:
document migration strategies and mayor changes in Plone that impact migrations

pbauer committed:
document migration strategies and mayor changes in Plone that impact migrations

pbauer committed:
call it Plone 6.0

pbauer committed:
call it Plone 6.0

stevepiercy committed:
Merge branch 'upgrade-guide' of github.com:plone/documentation into upgrade-guide

stevepiercy committed:
Rename files to spinal case
Add missing files and images

stevepiercy committed:
Add meta information, tidy.

stevepiercy committed:
Clean up upgrade introduction

stevepiercy committed:
Use sentence from intro.md

stevepiercy committed:
Clean up preparations.md

mauritsvanrees committed:
mayor -> major

mauritsvanrees committed:
Fixed link to version specific migration.

mauritsvanrees commented:
Thanks @pbauer !
I have started looking at this, and pushed two small changes.
Locally with `make livehtml` I see a few warnings about missing links, images and references. Do you have some uncommitted files locally? Especially `backend/upgrading/addon_upgrade`.

```
reading sources... [100%] backend/upgrading/version_specific_migration/upgrade_to_51                                    
/Users/maurits/community/plone-coredev/6.0/documentation/docs/docs/backend/upgrading/index.md:21: WARNING: toctree contains reference to nonexisting document 'backend/upgrading/addon_upgrade'
/Users/maurits/community/plone-coredev/6.0/documentation/docs/docs/backend/upgrading/version_specific_migration/upgrade_to_51.md:312: WARNING: image file not readable: _static/plip_hidpi_low_resolution.png
/Users/maurits/community/plone-coredev/6.0/documentation/docs/docs/backend/upgrading/version_specific_migration/upgrade_to_51.md:319: WARNING: image file not readable: _static/plip_hidpi_high_resolution.png
/Users/maurits/community/plone-coredev/6.0/documentation/docs/docs/backend/upgrading/version_specific_migration/upgrade_to_51.md:378: WARNING: image file not readable: _static/site-control-panel-icon-thumb-settings.png
looking for now-outdated files... none found
pickling environment... done
checking consistency... /Users/maurits/community/plone-coredev/6.0/documentation/docs/docs/volto/cheatsheet.md: WARNING: document isn't included in any toctree
/Users/maurits/community/plone-coredev/6.0/documentation/docs/docs/volto/releases.md: WARNING: document isn't included in any toctree
done
preparing documents... done
writing output... [100%] index                                                                                          
/Users/maurits/community/plone-coredev/6.0/documentation/docs/docs/backend/upgrading/version_specific_migration/upgrade_to_51.md:17: WARNING: unknown document: /develop/addons/upgrade_to_51
/Users/maurits/community/plone-coredev/6.0/documentation/docs/docs/backend/upgrading/version_specific_migration/upgrade_to_51.md:61: WARNING: undefined label: resource_bundle_aggregation
/Users/maurits/community/plone-coredev/6.0/documentation/docs/docs/backend/upgrading/version_specific_migration/upgrade_to_51.md:77: WARNING: undefined label: create_actions_ttw
/Users/maurits/community/plone-coredev/6.0/documentation/docs/docs/backend/upgrading/version_specific_migration/upgrade_to_51.md:134: WARNING: undefined label: addon_installation_code
/Users/maurits/community/plone-coredev/6.0/documentation/docs/docs/backend/upgrading/version_specific_migration/upgrade_to_51.md:256: WARNING: undefined label: using_external_catalogs
/Users/maurits/community/plone-coredev/6.0/documentation/docs/docs/backend/upgrading/version_specific_migration/upgrade_to_51.md:385: WARNING: undefined label: content-type-icons-changed
```

mauritsvanrees committed:
Improve upgrade preparation.

- Change indentation levels of the first list.
- Point to dist.plone.org instead of CMFPlone for the release notes.
- Remove instructions on first installing a newer Zope version. This was from the days before Zope and Plone were released as eggs.
- Add links to Zope documentation.
- Do not tell people to look at README/INSTALL/UPGRADE.txt in CMFPlone. I never update these.
- Fix backup link.

mauritsvanrees committed:
Improve upgrade troubleshooting.

- Fix some links.
- Products don't exist anymore. Use the term add-ons or packages.

mauritsvanrees committed:
Improve upgrade 5.2 to 6.0: 6.0 supports Py 3.8-3.10

mauritsvanrees commented:
I have pushed some more fixes to Preparation and Troubleshooting.

I wonder if we should include the old version specific information. We could include only the upgrade info from 5.2 to 6.0, and point to the old docs for older versions. But we should be sure that the old information remains available under some url. If the best way to ensure this is by including it here, that is fine with me.

The old docs have a page in `develop/addons` with specific information on [upgrading an add-on to 5.1](https://docs.plone.org/develop/addons/upgrade_to_51.html). I don't know why this was only done for 5.1, and why this is not in the main upgrade guide. We should probably merge this here.

stevepiercy committed:
Add html_meta tags

stevepiercy commented:
I'm also working on this guide. I'll pull in the commits from @mauritsvanrees, resolve, incorporate his comments, and push my work tonight. `make livehtml` reveals many b0rken things.

stevepiercy commented on docs/backend/upgrading/preparations.md:
```diff
@@ -16,54 +16,45 @@ Things to do before you migrate Plone.
-  - UPGRADE.txt (although this usually contains only the general procedure outlined above)
-
-    - These files are important because they may contain important last minute information and might be more specific than the relevant sections of this reference manual.
-
```
I could not find these files anywhere in https://dist.plone.org/release/6.0.0b2/, so I agree this is obsolete.

stevepiercy commented on docs/backend/upgrading/preparations.md:
```diff
@@ -16,54 +16,45 @@ Things to do before you migrate Plone.
-
-  - If Zope depends on a newer version of Python, install the new version of Python first.
-
-  - If the newer version of Plone depends on a newer version of Zope, you will need to install that before proceeding with the Plone upgrade.
```
I agree that we should not duplicate documentation of Zope, and together with the updated note providing better references to migration, I like this change.

stevepiercy committed:
Merge branch 'upgrade-guide' of github.com:plone/documentation into upgrade-guide

 Conflicts:
	docs/backend/upgrading/intro.md
	docs/backend/upgrading/preparations.md

pbauer committed:
fix errors and include old addon migration infos

pbauer commented:
I think having a upgrade-guide in the current docs that has info on the old versions and not only 5.2 to 6.0 is a good thing. Thus I added the old addon-upgrade-docs as well. But I don't want to force it so if you disagree feel free to remove these.

pbauer commented:
I think this can be merged for now.

stevepiercy commented:
Please wait until I complete my review. There is no rush. I have been kinda busy working on the installation docs, which take precedent.

stevepiercy committed:
Clean up intro.md

stevepiercy committed:
Fix grammar.
Condense list items. A nested list must have two blank lines before its start and after its end.

stevepiercy commented on docs/backend/upgrading/add-on-upgrade.md:
```diff
@@ -0,0 +1,37 @@
+    - You will find these in the documentation of each product.
+
+Should the `/prefs_install_products_form` be unreachable, you should try doing the add-on upgrades from the Management Interface.
+Navigate to the quickinstaller in the Management Interface, and reinstall or upgrade products that are shown to be outdated.
```
Where is this, "quickinstaller in the Management Interface"? I cannot find it anywhere in the ZMI.

stevepiercy replied:
@mauritsvanrees would you please take a look at commit ab70d94633b2e64e619f91f4b3c89d31a1baa124, and provide any feedback? Thank you!

mauritsvanrees replied:
Yes, that commit looks fine. Thanks.

stevepiercy committed:
Polish add-on-upgrade.md.

stevepiercy reviewed (commented):
@pbauer @mauritsvanrees or anyone else know where is "quickinstaller in the Management Interface"?

stevepiercy committed:
Add heading label

stevepiercy committed:
Clean up Troubleshooting

stevepiercy commented:
While reviewing content in this PR, I am discovering that there is incorrect and outdated information. I am adamant about publishing only correct and current information for Plone 6 documentation. That means reviews will take longer than expected, yet the content will be of higher quality. The steps of migrating docs by copy-paste and getting the docs to build without warnings and errors are always necessary and important, yet do not address this matter of quality.

stevepiercy committed:
Clean up version-specific migration procedures and tips index

stevepiercy committed:
Add keyword

stevepiercy committed:
Clean up code samples, add myst meta information

stevepiercy committed:
Update links
Fix grammar and syntax
Add PLIP to glossary

stevepiercy committed:
Clean up p4x-to-p5x-upgrade.md

stevepiercy committed:
Clean up upgrade-addons-to-50.md

stevepiercy committed:
Clean up plone5-minor-upgrade.md
Add html_meta header
Fix MyST and English grammar and syntax

stevepiercy committed:
Tidy upgrade-to-51.md

stevepiercy committed:
Tidy upgrade-addons-to-51.md

stevepiercy committed:
Tidy upgrade-addons-to-52.md
Backport sentence from 52 to 51.

stevepiercy committed:
Fix redirect links

stevepiercy committed:
Generic Setup -> `GenericSetup` because it is a package name

stevepiercy committed:
Remove useless heading, decrease heading levels
Update MyST meta tags

stevepiercy commented on docs/backend/upgrading/version-specific-migration/upgrade-to-python3.md:
````diff
@@ -0,0 +1,386 @@
+```{warning}
+The following section is valid until the final release of Plone 5.2.
+Upon the final release of Plone 5.2, something else will take its place.
+```
````
Can this warning be removed or replaced with whatever that "something" is?

mauritsvanrees replied:
The warning can be removed. I have no idea what to replace it with, and the documentation below seems okay still.

stevepiercy replied:
Removed in a7076aa98989067b994219c7884d99f29287adff

stevepiercy reviewed (commented):
@pbauer @mauritsvanrees or anyone else know where is "quickinstaller in the Management Interface"?

stevepiercy committed:
Tidy up upgrade-to-python3.md

stevepiercy committed:
Clean up upgrade-zodb-to-python3.md

stevepiercy committed:
Fix CSS snippet

stevepiercy committed:
Fix linkcheck

mauritsvanrees commented:
> @pbauer @mauritsvanrees or anyone else know where is "quickinstaller in the Management Interface"?

In Plone 5.2 and earlier it is in `portal_quickinstaller`.
In Plone 6.0, this tool is no longer there and its code is not included. One of the earliest upgrade steps in Plone 6.0 removes the tool if it is still there.

stevepiercy committed:
Remove obsolete warning

stevepiercy commented:
> In Plone 5.2 and earlier it is in `portal_quickinstaller`. In Plone 6.0, this tool is no longer there and its code is not included. One of the earliest upgrade steps in Plone 6.0 removes the tool if it is still there.

At https://demo.plone.org/portal_quickinstaller/manage_installProductsForm:

> WARNING: This form is deprecated. Please use the [Add-ons control panel](https://demo.plone.org/prefs_install_products_form) in the Plone UI.

The wording of this will be a little tricky, but let me see what I can whip together.

stevepiercy committed:
Deal with portal_quickinstaller deprecation and removal

stevepiercy committed:
Clean up upgrade-to-60.md

stevepiercy committed:
`GenericSetup` is an inline literal

stevepiercy committed:
Add Plone Classic UI landing page image

stevepiercy committed:
Minor clean up of migrate-to-volto.md

stevepiercy committed:
Merge branch '6-dev' into upgrade-guide

# Conflicts:
#	docs/conf.py
#	docs/glossary.md

stevepiercy committed:
Clean up migrate-to-volto.md

stevepiercy commented:
I have completed my review. I fixed mostly English and MyST syntax and grammar, some links and redirects, and a bunch of nit-picky details.

Now is the time for a final review. @pbauer @ksuess @mauritsvanrees

stevepiercy committed:
Merge branch '6-dev' into upgrade-guide

stevepiercy committed:
Update submodules/volto tip

pbauer commented:
LGTM

stevepiercy committed:
Merge branch '6-dev' into upgrade-guide

# Conflicts:
#	docs/glossary.md

stevepiercy committed:
Merge branch '6-dev' into upgrade-guide

stevepiercy reviewed (approved):
My changes were approved by @pbauer (one cannot approve a PR one creates).
